test/sort.jl

```diff
@@ -142,3 +142,46 @@ using InMemoryDatasets, PooledArrays, Random, Test, CategoricalArrays
     @test sortperm(dsl, :)==sortperm(dsr, :)
     @test sortperm(dsl, :, rev=true)==sortperm(dsr, :, rev=true)
 end
+
+@testset "_find_starts_of_groups" begin
```

Comment:
You need to add some tests where you have different type of data (and combination of them) such as PooledArray, CategoricalArray, Float64,...
consider cases where there are missing values in data
consider cases where you have a larger data set perhaps 10,000 observations with 10 columns/ with many levels/ with just a few levels

Reply:
OK, I will add more test cases.